action/action.rb

```diff
@@ -67,25 +57,46 @@ def file_fullpath(relative_path)
   end
 end
 
-rubocop_output.files.each do |file|
-  path = file_fullpath(file.path)
+def generate_annotations(compare_sha:)
+  annotations = []
+
+  rubocop_json = Bundler.with_original_env do
+    `git diff --name-only #{compare_sha} --diff-filter AM --relative | xargs rubocop --force-exclusion --format json`
+  end
+
+  rubocop_output = JSON.parse(rubocop_json, object_class: OpenStruct)
+
+  rubocop_output.files.each do |file|
+    path = file_fullpath(file.path)
 
-  change_ranges = GitUtils.generate_change_ranges(path, compare_sha: compare_sha)
+    change_ranges = GitUtils.generate_change_ranges(path, compare_sha: compare_sha)
 
-  file.offenses.each do |offense|
-    next unless change_ranges.any? { |range| range.include?(offense.location.start_line) }
+    file.offenses.each do |offense|
+      next unless change_ranges.any? { |range| range.include?(offense.location.start_line) }
 
-    annotations.push(
-      path: path,
-      start_line: offense.location.start_line,
-      end_line: offense.location.last_line,
-      annotation_level: RUBOCOP_TO_GITHUB_SEVERITY[offense.severity],
-      message: offense.message
-    )
+      annotations.push(
+        path: path,
+        start_line: offense.location.start_line,
+        end_line: offense.location.last_line,
+        annotation_level: RUBOCOP_TO_GITHUB_SEVERITY[offense.severity],
+        message: offense.message
+      )
+    end
   end
-end
 
-resp = check_run.update(annotations: annotations)
+  annotations
+end
 
-p resp
-p resp.json
+begin
+  annotations = generate_annotations(compare_sha: event.pull_request.base.sha)
+rescue Exception => e
+  puts e.message
+  puts e.backtrace.inspect
+  resp = check_run.error(message: e.message)
+  p resp
+  p resp.json
+else
+  resp = check_run.update(annotations: annotations)
+  p resp
```

Comment:
In the success case, what do you think about removing the extra debug output? Now that it's there in an error case, I'm not sure what this specific output offers